themes/robbyrussell/robbyrussell.theme.bash

```diff
@@ -18,8 +18,19 @@ function git_prompt_info() {
   echo -e "$SCM_PREFIX${bold_red}$SCM_BRANCH$SCM_STATE$SCM_SUFFIX"
 }
 
+function venv_prompt() {
+	python_venv=""
+	# Detect python venv
+	if [[ -n "${CONDA_DEFAULT_ENV}" ]]; then
+		python_venv="($PYTHON_VENV_CHAR${CONDA_DEFAULT_ENV}) "
+	elif [[ -n "${VIRTUAL_ENV}" ]]; then
+		python_venv="($PYTHON_VENV_CHAR$(basename "${VIRTUAL_ENV}")) "
+	fi
+	[[ -n "${python_venv}" ]] && echo "${python_venv}"
+}
+
 function prompt_command() {
-  PS1="${bold_green}➜  ${bold_cyan}\W${reset_color}$(scm_prompt_info)${normal} "
+  PS1="$(venv_prompt)${bold_green}➜  ${bold_cyan}\W${reset_color}$(scm_prompt_info)${normal} "
```

Comment:
It _feels_ a little too opinionated to place the venv_prompt even **before** the `➜` ...  not sure if _before_ scm_prompt_info or _after_ it is better, but I'm leaning toward after ...

Reply:
Hi, venv_prompt was usually on the leftmost side of the P1 by default with virtualenv, pipenv, and pyenv. `robbyrussell` theme from oh-my-zsh implements it also on the leftmost side before the `➜`.

Reply:
@davidpfarrell Is there another way to modify the prompt of the theme, without changing the original theme? Because, original theme is not showing virtualenv prompt, can it be overriden through the `.bashrc`? Or leave somehow up to the underlying venv to modify prompt with their default (but theme seems to just ignores venv modifications of PS1 - don't know why).

Reply:
@petarnikolovski, as it stands now most themes overwrite `$PS1` before displaying every prompt. Modifications to `$PS1` are just entirely lost immediately after setting them. Something like the PowerLine theme(s) allow for a *lot* more fine tuning, but that requires using one of those themes.

Reply:
@davidpfarrell I've checked virtualenv activate script, and it places the prompt on the leftmost side:

```sh
mkdir /tmp/project
cd /tmp/project
python3 -m venv env
cat env/bin/activate

...
if [ -z "${VIRTUAL_ENV_DISABLE_PROMPT:-}" ] ; then
    _OLD_VIRTUAL_PS1="${PS1:-}"
    if [ "x(env) " != x ] ; then
	PS1="(env) ${PS1:-}"
    else
    if [ "`basename \"$VIRTUAL_ENV\"`" = "__" ] ; then
        # special case for Aspen magic directories
        # see http://www.zetadev.com/software/aspen/
        PS1="[`basename \`dirname \"$VIRTUAL_ENV\"\``] $PS1"
    else
        PS1="(`basename \"$VIRTUAL_ENV\"`)$PS1"
    fi
    fi
    export PS1
fi
...
```